client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/BooleanClassReplacement.java

```diff
@@ -0,0 +1,50 @@
+package org.evomaster.client.java.instrumentation.coverage.methodreplacement.classes;
+
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.DistanceHelper;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.MethodReplacementClass;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.Replacement;
+import org.evomaster.client.java.instrumentation.heuristic.Truthness;
+import org.evomaster.client.java.instrumentation.shared.ReplacementType;
+import org.evomaster.client.java.instrumentation.shared.StringSpecialization;
+import org.evomaster.client.java.instrumentation.shared.StringSpecializationInfo;
+import org.evomaster.client.java.instrumentation.shared.TaintInputName;
+import org.evomaster.client.java.instrumentation.staticstate.ExecutionTracer;
+
+import static org.evomaster.client.java.instrumentation.coverage.methodreplacement.DistanceHelper.H_REACHED_BUT_NULL;
+
+public class BooleanClassReplacement implements MethodReplacementClass {
+
+    @Override
+    public Class<?> getTargetClass() {
+        return Boolean.class;
+    }
+
+    @Replacement(type = ReplacementType.BOOLEAN, replacingStatic = true)
+    public static boolean parseBoolean(String input, String idTemplate) {
+
+        if (TaintInputName.isTaintInput(input)) {
+            ExecutionTracer.addStringSpecialization(input,
+                    new StringSpecializationInfo(StringSpecialization.INTEGER, null));
```

Comment:
should rather create a `StringSpecialization.BOOLEAN`, which I can then implement. 
could add a TODO, and then I ll fix it

Reply:
ok. I also added the translation to BooleanGene.